purchases/src/main/kotlin/com/revenuecat/purchases/SyncPurchasesHelper.kt

```diff
@@ -0,0 +1,100 @@
+package com.revenuecat.purchases
+
+import com.revenuecat.purchases.common.BillingAbstract
+import com.revenuecat.purchases.common.LogIntent
+import com.revenuecat.purchases.common.ReceiptInfo
+import com.revenuecat.purchases.common.debugLog
+import com.revenuecat.purchases.common.errorLog
+import com.revenuecat.purchases.common.log
+import com.revenuecat.purchases.identity.IdentityManager
+import com.revenuecat.purchases.interfaces.ReceiveCustomerInfoCallback
+import com.revenuecat.purchases.models.StoreTransaction
+import com.revenuecat.purchases.strings.PurchaseStrings
+
+internal class SyncPurchasesHelper(
+    private val billing: BillingAbstract,
+    private val identityManager: IdentityManager,
+    private val customerInfoHelper: CustomerInfoHelper,
+    private val postReceiptHelper: PostReceiptHelper
+) {
+    fun syncPurchases(
+        isRestore: Boolean,
+        onSuccess: (CustomerInfo) -> Unit,
+        onError: (PurchasesError) -> Unit
+    ) {
+        log(LogIntent.DEBUG, PurchaseStrings.SYNCING_PURCHASES)
+
+        val appUserID = identityManager.currentAppUserID
+
+        billing.queryAllPurchases(
+            appUserID,
+            onReceivePurchaseHistory = { allPurchases ->
+                if (allPurchases.isNotEmpty()) {
+                    val lastPurchase = allPurchases.last()
+                    val errors: MutableList<PurchasesError> = mutableListOf()
+                    fun handleLastPurchase(currentPurchase: StoreTransaction, lastPurchase: StoreTransaction) {
+                        if (currentPurchase == lastPurchase) {
+                            if (errors.isEmpty()) {
+                                debugLog(PurchaseStrings.SYNCED_PURCHASES_SUCCESSFULLY)
+                                retrieveCustomerInfoFromCache(appUserID, onSuccess, onError)
```

Comment:
That window could exist if calling syncPurchases right after configure? If there are no purchases and no cached info has been retrieved. What's the problem with doing a CacheFetchPolicy `CACHED_OR_FETCHED` to prevent that from happening.

Reply:
Hmm rethinking about it, I don't think it would be an issue... It might result in extra fetches when developers use `syncPurchases`, but it shouldn't be a big issue. Will change that